src/macrofier/engine.mjs

```diff
@@ -486,29 +488,77 @@ function insertTableofContents(content) {
+        enm.title = k
+      } else if (v.type === 'object') {
+        enm = getEnumProperties(v)
+      } else if (v.type === 'array') {
+        enm = Object.assign({}, (v.items[0] ? v.items[0] : v.items))
+        enm.title = k
+      }
+      return enm
+    })),
+    map(Object.entries),
+    filter(schema => isObject(schema))
+)(schema)
+
+const convertEnumTemplate = (sch, templateName, templates) => {
+  const template = getTemplate(templateName, templates).split('\n')
+  let schema = isEnumType(sch) ? sch : getEnumProperties(sch)
+  for (var i = 0; i < template.length; i++) {
+    if (template[i].indexOf('${key}') >= 0) {
+      template[i] = schema.enum.map(value => {
+        const safeName = value.split(':').pop().replace(/[\.\-]/g, '_').replace(/\+/g, '_plus').replace(/([a-z])([A-Z0-9])/g, '$1_$2').toUpperCase()
+        return template[i].replace(/\$\{key\}/g, safeName)
+                          .replace(/\$\{value\}/g, value)
+      }).join('\n')
+      if (!templateName.includes(".cpp")) {
+        template[i] = template[i].replace(/,*$/, '');
+      }
+    }
+  }
+  return template.join('\n')
+                 .replace(/\$\{title\}/g, capitalize(schema.title))
+                 .replace(/\$\{description\}/g, schema.description ? ('- ' + schema.description) : '')
+                 .replace(/\$\{name\}/g, schema.title)
+                 .replace(/\$\{NAME\}/g, schema.title.toUpperCase())
+}
+
 const enumFinder = compose(
-  filter(x => x.type === 'string' && Array.isArray(x.enum) && x.title),
+  filter(x => ((isEnumType(x) && x.title) || isEnumProperties(x))),
   map(([_, val]) => val),
   filter(([_key, val]) => isObject(val))
 )
 
-const generateEnums = (json, templates) => {
+const generateEnums = (json, templates, options = { destination: '' }) => {
+  const suffix = options.destination.split('.').pop()
   return compose(
     option(''),
+    map(val => val ? (suffix && getTemplate(`/sections/enum.${suffix}`, templates) ? getTemplate(`/sections/enum.${suffix}`, templates).replace(/\$\{schema.list\}/g, val.trimEnd()) : val) : ''),
     map(reduce((acc, val) => acc.concat(val).concat('\n'), '')),
-    map(map((schema) => {
-      const template = getTemplate('/types/enum', templates).split('\n')
-      for (var i = 0; i < template.length; i++) {
-        if (template[i].indexOf('${key}') >= 0) {
-          template[i] = schema.enum.map(value => {
-            const safeName = value.split(':').pop().replace(/[\.\-]/g, '_').replace(/\+/g, '_plus').replace(/([a-z])([A-Z0-9])/g, '$1_$2').toUpperCase()
-            return template[i].replace(/\$\{key\}/g, safeName)
-                              .replace(/\$\{value\}/g, value)
-          }).join('\n')
-        }
-      }
-      return template.join('\n').replace(/\$\{name\}/g, schema.title).replace(/\$\{NAME\}/g, schema.title.toUpperCase())
-    })),
+    map(map((schema) => convertEnumTemplate(schema, suffix ? `/types/enum.${suffix}` : '/types/enum', templates))),
```

Comment:
This looks good, but the new files `/types/enum.cpp` and `/types/enum.h` are missing in the PR